test/unit/storage_async_test.rb

```diff
@@ -302,10 +338,42 @@ def assert_sentinel_config(conf, conn)
       host, port = endpoint.address
       assert_equal conf[:sentinels][i][:host], host
       assert_equal conf[:sentinels][i][:port], port
-      assert_equal(conf[:sentinels][i][:password], password) if conf[:sentinels][i].key? :password
     end unless conf[:sentinels].empty?
   end
 
+  def assert_acl_credentials(conf, client)
+    unless conf[:username].to_s.strip.empty? && conf[:password].to_s.strip.empty?
+      assert_instance_of ThreeScale::Backend::AsyncRedis::Protocol::ExtendedRESP2, client.protocol
+      username, password = client.protocol.instance_variable_get(:@credentials)
+      assert_equal conf[:username], username
+      assert_equal conf[:password], password
+    end
+
+    unless conf[:sentinel_username].to_s.strip.empty? && conf[:sentinel_password].to_s.strip.empty?
+      sentinel_username, sentinel_password = client.instance_variable_get(:@sentinel_credentials)
+      assert_equal conf[:sentinel_username], sentinel_username
+      assert_equal conf[:sentinel_password], sentinel_password
+    end
+  end
+
+  def assert_tls_certs(conf, client, test_cert_type)
+    unless conf[:ssl_params].to_s.strip.empty?
```

Comment:
That configuration would make no sense, I think it's fine the connection fails in that situation.

Reply:
oh wait, that situation does make sense indeed. If they set `ssl: true` but provide no certs, then the certs from the system should be used by default. That would surely work in sync mode, but I'm not sure it's implemented in async mode. Let me take a look.

Reply:
I've been investigating and I don't know how could I test this scenario. I would need to have a cert signed by one of the well-known CAs which certificates are installed in fedora, and also the private key corresponding to that cert. Then configure my local redis server to use that certificate and try to connect from apisonator with `ssl: true` but no `ssl_params`.

I can't test it myself but my guess is it would work fine, also in async mode, because we rely on `Async::IO::SSLEndpoint.new`.

Reply:
well, we just test that the configuration is as expected I guess.

Reply:
btw if we really want to test scerario where OS CA should be used, we can setup /etc/pki with fakefs with custom certificates. But I think we don't need to test ruby implementation.

Reply:
> btw if we really want to test scerario where OS CA should be used, we can setup /etc/pki with fakefs with custom certificates.

Yeah, that's right.

> But I think we don't need to test ruby implementation.

Yeah, that's right.